### PR TITLE
Removed testing for delegates in different running modes from iOS Vision Tasks

### PR DESCRIPTION
**Pull request by priankakariatyml**

### CODE REVIEW

File: mediapipe/tasks/ios/core/sources/MPPTaskRunner.mm [hunk 1]
```diff
@@ -59,7 +62,20 @@ - (instancetype)initWithCalculatorGraphConfig:(CalculatorGraphConfig)graphConfig
 }
 
 - (BOOL)sendPacketMap:(const PacketMap &)packetMap error:(NSError **)error {
+  if (!_initializedWithPacketsCallback) {
+    [MPPCommonUtils
+        createCustomError:error
+                 withCode:MPPTasksErrorCodeFailedPreconditionError
+              description:[NSString
+                              stringWithFormat:@"This method can only be called if the task is "
+                                               @"running in a stream mode and an object of a class "
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/core/sources/MPPTaskRunner.mm:71]: "an object of a class" is a bit hard to parse. Maybe just "and a delegate is provided in the task options..."

File: mediapipe/tasks/ios/core/sources/MPPTaskRunner.mm [hunk 2]
```diff
@@ -59,7 +62,20 @@ - (instancetype)initWithCalculatorGraphConfig:(CalculatorGraphConfig)graphConfig
 }
 
 - (BOOL)sendPacketMap:(const PacketMap &)packetMap error:(NSError **)error {
+  if (!_initializedWithPacketsCallback) {
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/core/sources/MPPTaskRunner.mm:65]: This now means that we can initialize a task that will only ever throw errors. Is it possible to revert to the old behavior and throw an error on the initialization?

Comment (priankakariatyml) [reply mediapipe/tasks/ios/core/sources/MPPTaskRunner.mm:65]: Yeah, but then this whole change is unnecessary right? I mean wouldn't it suffice to just make all the delegate methods @required. Can I close this PR ?

Comment (schmidt-sebastian) [reply mediapipe/tasks/ios/core/sources/MPPTaskRunner.mm:65]: Probably. I think we can have `@required` methods and tests that the delegate object is set itself (like I believe we do today).
